will/plugin.py

```diff
@@ -83,54 +89,82 @@ def say(self, content, message=None, room=None, channel=None, service=None, pack
             )
             if package_for_scheduling:
                 return "message.outgoing.%s" % backend, e
-            else:
-                logging.info("putting in queue: %s" % content)
-                self.publish("message.outgoing.%s" % backend, e)
+            logging.info("putting in queue: %s", content)
+            self.publish("message.outgoing.%s" % backend, e)
+        return None
+
+    def send_file(self, message, content, filename, text=None, filetype='text',
```

Comment:
Let's split file upload to a separate PR as I know at least a few of us have written our own implementation (for example #372 )